### PR TITLE
8253935: [testbug] ComboBoxTest.testEditorKeyInputsWhenPopupIsShowing fails on Mac, Linux

### PR DESCRIPTION
**Pull request by arapte**
The test fails on Mac and Linux.
Cause of failure:
- Linux: Unlike Windows, on Linux CTRL + Arrow keys move the caret to end/beginning of current word and not to beginning/end of next word.
- Mac: On Mac correct keys to traverse through words is ALT + Arrow and not CTRL + Arrow. And like on Linux, ALT + Arrow keys move the caret to end/beginning of current word and not to beginning/end of next word.

Fix is to use correct keys and asserts according to platform.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253935](https://bugs.openjdk.java.net/browse/JDK-8253935): [testbug] ComboBoxTest.testEditorKeyInputsWhenPopupIsShowing fails on Mac, Linux

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/317/head:pull/317`
`$ git checkout pull/317`

### CODE REVIEW

**Review by kevinrushforth — COMMENTED** [review]
The test changes look good, and pass for me on all three platforms. I left a couple suggestions for you to consider.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java [hunk 1]
```diff
@@ -1384,27 +1385,46 @@ public void defaultConverterCanHandleIncorrectType_2() {
...
 
-        // Test CTRL + RIGHT key
-        keyboard.doKeyPress(KeyCode.RIGHT, KeyModifier.CTRL);
+        if (Utils.isMac()) {
+            // Test ALT + RIGHT key
+            keyboard.doKeyPress(KeyCode.RIGHT, KeyModifier.ALT);
+        } else {
+            // Test CTRL + RIGHT key
+            keyboard.doKeyPress(KeyCode.RIGHT, KeyModifier.getShortcutKey());
+        }
```
[the remark this answers is not in the document]
Comment (arapte) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java]: That seems better, added `getWordTraversalKey()` method to `KeyModifier`.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java [hunk 2]
```diff
@@ -1384,27 +1385,46 @@ public void defaultConverterCanHandleIncorrectType_2() {
...
         assertEquals("", cb.getEditor().getSelectedText());
-        assertEquals(4, cb.getEditor().getCaretPosition());
+        if (Utils.isWindows()) {
+            assertEquals(4, cb.getEditor().getCaretPosition());
+        } else {
+            assertEquals(3, cb.getEditor().getCaretPosition());
+        }
```
[the remark this answers is not in the document]
Comment (arapte) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java:1392]: Changed both the asserts as you mentioned.

**Review by kevinrushforth — APPROVED** [review]
Looks good.

**Pull request closed without merging**